### PR TITLE
add AWS Load Balancer Controller blog

### PR DESCRIPTION
**Pull request by rubyrat**
adding blog - AWS Load Balancer Controller with existing ALB

### CODE REVIEW

File: src/posts/aws-load-balancer-controller-with-existing-alb.md [hunk 1]
```diff
@@ -0,0 +1,378 @@
...
+metadata:
+  name: ingress-nginx-binding
+spec:
+  # The service we want to connect to.
+  # We use the Ingress Nginx Controller so let's point to that service.
+  # NOTE: it was necessary for this TargetGroupBinding to be in the same namespace as the service.
+  serviceRef:
+    name: ingress-nginx
+    port: 80
+	# NOTE: need the ARN of the TargetGroup... Bit of a PITA.
```
Comment (jlarfors) [inline src/posts/aws-load-balancer-controller-with-existing-alb.md:63]: This looks like a tab, and is not aligned with the line below...

File: src/posts/aws-load-balancer-controller-with-existing-alb.md [hunk 2]
```diff
@@ -0,0 +1,378 @@
...
+
+Let’s look at how we implemented this with the AWS Load Balancer Controller next.
+
+## Implementation
+
+### Terraform
+
+We use Terraform to manage (amongst other things) the EKS clusters, ALBs and TargetGroups. For implementing the AWS Load Balancer Controller all we needed to do was create the necessary IAM role that can be assumed by a Kubernetes ServiceAccount. The following snippet show this:
+
+```
```
Comment (jlarfors) [inline src/posts/aws-load-balancer-controller-with-existing-alb.md:84]: hcl code block

File: src/posts/aws-load-balancer-controller-with-existing-alb.md [hunk 3]
```diff
@@ -0,0 +1,378 @@
...
+
+The `TargetGroupBinding` Kubernetes Custom Resource we need to create requires the TargetGroup ARN which is non deterministic. In our setup, we use Terraform to create the [Kubernetes secret](https://argo-cd.readthedocs.io/en/stable/operator-manual/declarative-setup/#clusters) that informs ArgoCD about connected clusters. Within that secret we can attach additional labels that can be accessed by the ArgoCD [ApplicationSets](https://argocd-applicationset.readthedocs.io/en/stable/), which means we have a very primitive way of passing data from Terraform to ArgoCD without ay extra tools. Note that Kubernetes labels have some restrictions on the [syntax and character set](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set) that can be used, so we can’t just pass in arbitrary data.
+
+Here is how we create the Kubernetes secret that essentially “register” a cluster with ArgoCD, and how we pass the TargetGroup name and ID via labels.
+
+```
```
Comment (jlarfors) [inline src/posts/aws-load-balancer-controller-with-existing-alb.md:147]: `hcl`

File: src/posts/aws-load-balancer-controller-with-existing-alb.md [hunk 4]
```diff
@@ -0,0 +1,378 @@
...
+```yaml
+# File: appset-aws-lb-controller.yaml
+apiVersion: argoproj.io/v1alpha1
+kind: ApplicationSet
+metadata:
+  name: aws-lb-controller
+  namespace: argocd
+spec:
+  generators:
+		# This is a little trick we use for nearly all our apps to control which apps
```
Comment (jlarfors) [inline src/posts/aws-load-balancer-controller-with-existing-alb.md:245]: Looks like tabs again, should be spaces and aligned with the lines below.

File: src/posts/aws-load-balancer-controller-with-existing-alb.md [hunk 5]
```diff
@@ -0,0 +1,378 @@
...
+        helm:
+          releaseName: aws-lb-controller
+          values: |
+            clusterName: {{ name }}
+            serviceAccount:
+              create: true
+              annotations:
+                "eks.amazonaws.com/role-arn": "arn:aws:iam::123456789012:role/{{ name }}-aws-lb-controller"
+              name: aws-lb-controller
+						# We won't be using ingresses with this controller.
```
Comment (jlarfors) [inline src/posts/aws-load-balancer-controller-with-existing-alb.md:285]: tab

File: src/posts/aws-load-balancer-controller-with-existing-alb.md [hunk 6]
```diff
@@ -0,0 +1,378 @@
...
+kind: TargetGroupBinding
+metadata:
+  name: ingress-nginx
+spec:
+  targetType: instance
+  serviceRef:
+    name: ingress-nginx-controller
+    port: 80
+  targetGroupARN: {{ required "targetGroupArn required" .Values.targetGroupArn }}
+	# By default, add all nodes to the cluster unless they have the label
```
Comment (jlarfors) [inline src/posts/aws-load-balancer-controller-with-existing-alb.md:361]: tab

File: src/posts/aws-load-balancer-controller-with-existing-alb.md [hunk 7]
```diff
@@ -0,0 +1,378 @@
...
+	# exclude-from-lb-targetgroups set
```
Comment (jlarfors) [inline src/posts/aws-load-balancer-controller-with-existing-alb.md:362]: tab

**Review by jlarfors — CHANGES_REQUESTED** [review]

File: src/posts/aws-load-balancer-controller-with-existing-alb.md (hunk 2 above)
Comment (rubyrat) [reply src/posts/aws-load-balancer-controller-with-existing-alb.md:84]: fixed

File: src/posts/aws-load-balancer-controller-with-existing-alb.md (hunk 3 above)
Comment (rubyrat) [reply src/posts/aws-load-balancer-controller-with-existing-alb.md:147]: fixed

File: src/posts/aws-load-balancer-controller-with-existing-alb.md (hunk 1 above)
Comment (rubyrat) [reply src/posts/aws-load-balancer-controller-with-existing-alb.md:63]: Screenshot 2022-10-17 at 16 10 13
This is what it looks like on my local copy. It looks aligned, so I'm not sure how to fix it (same with the other tab issues).

**Review by jlarfors — APPROVED** [review]
LGTM!

**Pull request merged**